src/parsers/power_models_data.jl

```diff
@@ -162,6 +162,105 @@ function _get_pm_3w_name(
     return "$(get_name(bus_primary))-$(get_name(bus_secondary))-$(get_name(bus_tertiary))-i_$ckt"
 end
 
+"""
+Parses ITC data from a dictionary and constructs a lookup table
+of piecewise linear scaling functions.
+"""
+function _impedance_correction_table_lookup(data::Dict)
+    pwl_lookup = Dict{Int64, PiecewiseLinearData}()
+    type_lookup = Dict{Int64, Int64}()
+    ict_instances = Dict{Tuple{Int64, Int}, ImpedanceCorrectionData}()
+
+    @info "Reading Impedance Correction Table data"
+    if !haskey(data, "impedance_correction")
+        @info "There is no Impedance Correction Table data in this file"
+        return
+    end
+
+    for (table_number, table_data) in data["impedance_correction"]
+        table_number = parse(Int64, table_number)
+        x = table_data["tap_or_angle"]
+        y = table_data["scaling_factor"]
+
+        if length(x) == length(y)
+            pwl_data = PiecewiseLinearData([(x[i], y[i]) for i in eachindex(x)])
+            pwl_lookup[table_number] = pwl_data
+
+            table_type =
+                if (x[1] >= TAP_RATIO_LBOUND && x[1] <= TAP_RATIO_UBOUND)
+                    TAP_RATIO
```

Comment:
yeah there's no reason to use numbers rather than enum values here, that's what enums are for